core-v1/src/main/scala/data/Justified.scala

```diff
@@ -258,6 +258,40 @@ object Justified extends LowPriorityJustifiedImplicits {
     override def wrapFact[O](fact: TypedFact[O])(implicit opO: Any): Justified[O] = Justified.byFact(fact)
   }
 
+  implicit def wrapContained[OP[_]]: WrapContained[Justified, OP] =
+    WrapContainedJustified.asInstanceOf[WrapContained[Justified, OP]]
+
+  private final case object WrapContainedJustified extends WrapContained[Justified, Any] {
+
+    override def wrapContained[C[_] : Foldable, V](
+      original: C[Justified[V]],
+      valid: Set[Justified[V]],
+      found: Seq[Justified[V]],
+    )(implicit
+      opV: Any,
+    ): Justified[Boolean] = {
+      // TODO: Figure out how to avoid the Justified.byConst situation.
+      // One option is to wrap every collection as well as the elements to provide traceability for empty collections
+      val justifiedValidValues = NonEmptySeq
+        .fromSeq(valid.toSeq)
+        .map { nonEmpty =>
+          Justified.byInference("validValues", valid.map(_.value), nonEmpty)
+        }
+        .getOrElse {
+          Justified.byConst(Set())
+        }
```

Comment:
I think this can be addressed using a wrapper around a collection of wrapped elements. This would require a significant refactor across the DSL, but I think it should be done pre-1.0.0.